### PR TITLE
[sp] improve data table header

### PR DESCRIPTION
**Pull request by shrey-mage**
# Summary
- add column links to data table header
- fix pie chart glitch when there are very few columns

# Tests
Use custom actions to remove most of a dataset's columns. The pie chart should no longer glitch out and maintain the same "donut hole" size.

image
image

cc: @johnson-mage @wangxiaoyou1993

### CODE REVIEW

File: mage_ai/frontend/components/datasets/overview/utils.tsx [hunk 1]
```diff
@@ -336,7 +341,24 @@ export function buildRenderColumnHeader({
...
             }}
           >
-            {columns[columnIndex]}
+            <NextLink
+              as={createDatasetTabRedirectLink(TAB_REPORTS, columnIndex)}
+              href="/datasets/[...slug]"
+              passHref
+            >
+              <Link>
+                <Text
```
Comment (johnson-mage) [inline mage_ai/frontend/components/datasets/overview/utils.tsx:350]: is this Text component necessary?

File: mage_ai/frontend/components/datasets/overview/utils.tsx [hunk 2]
```diff
@@ -336,7 +341,24 @@ export function buildRenderColumnHeader({
...
+                  backgroundColor={light.feature.active}
```
Comment (johnson-mage) [inline mage_ai/frontend/components/datasets/overview/utils.tsx:351]: does this do anything?

File: mage_ai/frontend/components/datasets/overview/utils.tsx [hunk 3]
```diff
@@ -18,8 +21,10 @@ import {
 import { COLUMN_TYPE_ICON_MAPPING } from '@components/constants';
 import { ChartTypeEnum } from '@interfaces/InsightsType';
 import { ColumnTypeEnum } from '@interfaces/FeatureType';
+import { TAB_REPORTS } from '.';
```
Comment (johnson-mage) [inline mage_ai/frontend/components/datasets/overview/utils.tsx:24]: perhaps update the import to `./index` for clarity

**Review by johnson-mage — APPROVED** [review]
approved after addressing comments

File: mage_ai/frontend/components/datasets/overview/utils.tsx (hunk 1 above)
Comment (shrey-mage) [reply mage_ai/frontend/components/datasets/overview/utils.tsx:350]: will fix in another PR

File: mage_ai/frontend/components/datasets/overview/utils.tsx (hunk 2 above)
Comment (shrey-mage) [reply mage_ai/frontend/components/datasets/overview/utils.tsx:351]: will fix in another PR

**Pull request merged**